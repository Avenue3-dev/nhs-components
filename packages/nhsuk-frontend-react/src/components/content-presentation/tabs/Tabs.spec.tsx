import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { Tabs } from './Tabs';

it('should render the Tabs component', async () => {
  const { container } = render(
    <Tabs>
      <Tabs.Title>Contents</Tabs.Title>
      <Tabs.List>
        <Tabs.ListItem href="#past-day">Past day</Tabs.ListItem>
        <Tabs.ListItem href="#past-week">Past week</Tabs.ListItem>
        <Tabs.ListItem href="#past-month">Past month</Tabs.ListItem>
        <Tabs.ListItem href="#past-year">Past year</Tabs.ListItem>
      </Tabs.List>
      <Tabs.Panel id="past-day">
        <h2>Past day</h2>
        <p>Content for the past day</p>
      </Tabs.Panel>
      <Tabs.Panel id="past-week">
        <h2>Past week</h2>
        <p>Content for the past week</p>
      </Tabs.Panel>
      <Tabs.Panel id="past-month">
        <h2>Past month</h2>
        <p>Content for the past month</p>
      </Tabs.Panel>
      <Tabs.Panel id="past-year">
        <h2>Past year</h2>
        <p>Content for the past year</p>
      </Tabs.Panel>
    </Tabs>,
  );

  expect(container).toMatchSnapshot();

  const secondTab = container.querySelector('[href="#past-week"]');

  await userEvent.click(secondTab!);

  expect(
    container.querySelector('[href="#past-week"]')?.parentElement,
  ).toHaveClass('nhsuk-tabs__list-item--selected');
});
