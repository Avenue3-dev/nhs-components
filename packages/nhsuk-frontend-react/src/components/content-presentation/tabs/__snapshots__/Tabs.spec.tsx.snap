// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Tabs component 1`] = `
<div>
  <div
    class="nhsuk-tabs"
    data-module="nhsuk-tabs"
  >
    <h2
      class="nhsuk-tabs__title"
    >
      Contents
    </h2>
    <ul
      class="nhsuk-tabs__list"
    >
      <li
        class="nhsuk-tabs__list-item"
      >
        <a
          class="nhsuk-tabs__tab"
          href="#past-day"
        >
          Past day
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
      >
        <a
          class="nhsuk-tabs__tab"
          href="#past-week"
        >
          Past week
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
      >
        <a
          class="nhsuk-tabs__tab"
          href="#past-month"
        >
          Past month
        </a>
      </li>
      <li
        class="nhsuk-tabs__list-item"
      >
        <a
          class="nhsuk-tabs__tab"
          href="#past-year"
        >
          Past year
        </a>
      </li>
    </ul>
    <div
      class="nhsuk-tabs__panel"
      id="past-day"
    >
      <h2>
        Past day
      </h2>
      <p>
        Content for the past day
      </p>
    </div>
    <div
      class="nhsuk-tabs__panel"
      id="past-week"
    >
      <h2>
        Past week
      </h2>
      <p>
        Content for the past week
      </p>
    </div>
    <div
      class="nhsuk-tabs__panel"
      id="past-month"
    >
      <h2>
        Past month
      </h2>
      <p>
        Content for the past month
      </p>
    </div>
    <div
      class="nhsuk-tabs__panel"
      id="past-year"
    >
      <h2>
        Past year
      </h2>
      <p>
        Content for the past year
      </p>
    </div>
  </div>
</div>
`;
